celeryutils/persist_on_failure.py

```diff
@@ -0,0 +1,52 @@
+"""
+Celery utility code for persistent tasks.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from celery import Task
+
+from .models import FailedTask
+
+
+# pylint: disable=abstract-method
+class PersistOnFailureTask(Task):
+    """
+    Custom Celery Task base class that persists task data on failure.
+    """
+
+    abstract = True
+    typing = False
```

Comment:
In Celery 3, celery will blindly send the arguments you pass to your command, whether or not they are the right arguments.  The task will then fail if it can't handle them.  Celery 4 checks the arguments before it sends the task, and raises a TypeError if they don't check out.  This opts out of that behavior on Celery 4 (and has no effect on Celery 3). 

Submitting incorrectly called tasks allows us to to catch and persist those tasks in our FailedTask table.  Without this, the TypeError would occur on the app server, not the task worker, so the task would never get persisted.